### PR TITLE
improve+reduce fails+warnings in read+write functions; fix sorting order in quitteSort 

### PR DESCRIPTION
**Pull request by orichters**
1. If somehow an empty data.frame was written to a mif file and then read with read.quitte, the error message was incomprehensible:
```
Error in `if (last(period.columns) != length(header)) ...`:
! missing value where TRUE/FALSE needed
Run `rlang::last_trace()` to see where the error occurred.
```
Now, fail with
```
No column name found that could be understood as a 4-digit year.
```

2. let write.mif and write.IAMCxlsx try to convert to quitte instead of failing immediately, and warn in case of empty data

3. sort order in quitteSort according to function description, was somehow changed unnoticed in [this commit](https://github.com/pik-piam/quitte/commit/08450834a8fb67bb796cd190f22d4b25ce7624e7)

### CODE REVIEW

File: R/read.quitte.R [hunk 1]
```diff
@@ -129,6 +129,10 @@ read.quitte <- function(file,
             stop("missing default columns in header of file ", f, ": ",
                  paste(setdiff(default.columns, header[1:5]), collapse = ", "))
 
+        if (length(period.columns) == 0) {
+            stop("No column name found that could be understood as a 4-digit year.")
```
Comment (0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q) [inline R/read.quitte.R:133]: Please indicate the file which failed to read, as `read.quitte()` can read several files at once.

File: R/quitteSort.R [hunk 2]
```diff
@@ -20,6 +20,6 @@ quitteSort <- function(x) {
   }
 
   x %>%
-    arrange(.data$model, .data$scenario, .data$region, .data$variable,
-            .data$unit, .data$period)
+    arrange(.data$model, .data$scenario, .data$variable,
+            .data$unit, .data$region, .data$period)
```
Comment (0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q) [inline R/quitteSort.R:24]: NB: `unit` is superfluous here, as variables can have only one unit by definition.

File: R/write.IAMCxlsx.R [hunk 3]
```diff
@@ -22,19 +22,13 @@
 
 #' @export
 write.IAMCxlsx <- function(x, path, append = FALSE) {
-
+    x <- as.quitte(x)
+    if (nrow(x) == 0) warning("Empty data frame written to ", path)
```
Comment (0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q) [inline R/write.IAMCxlsx.R:26]: Interesting choice to report success at the head of a function rather then the bottom …

**Review by 0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q — APPROVED** [review]

File: R/quitteSort.R (hunk 2 above)
Comment (orichters) [reply R/quitteSort.R:24]: Only as long as you don't merge `GDP (billion US$2010/yr)` and `GDP (billion US$2005/yr)` data into one dataframe. Let us keep it.

**Pull request merged**